libPhoneNumber/NBPhoneNumberUtil+ShortNumber.m

```diff
@@ -0,0 +1,323 @@
+//
+//  NBPhoneNumberUtil+ShortNumber.m
+//  libPhoneNumberiOS
+//
+//  Created by Paween Itthipalkul on 11/29/17.
+//  Copyright © 2017 Google LLC. All rights reserved.
+//
+
+#import "NBPhoneNumberUtil+ShortNumber.h"
+
+#import <Foundation/Foundation.h>
+
+#import "NBMetadataHelper.h"
+#import "NBPhoneMetaData.h"
+#import "NBPhoneNumber.h"
+#import "NBPhoneNumberDesc.h"
+#import "NBPhoneNumberUtil+Category.h"
+#import "NBRegExMatcher.h"
+#import "NBRegularExpressionCache.h"
+
+#if SHORT_NUMBER_SUPPORT
+
+static NSString * const PLUS_CHARS_PATTERN = @"[+\uFF0B]+";
+
+@interface NBPhoneNumberUtil()
+
+@property (nonatomic) NSDictionary<NSNumber *, NSArray<NSString *> *> *countryToRegionCodeMap;
+
+@end
+
+@implementation NBPhoneNumberUtil (ShortNumber)
```

Comment:
Splitting this out to a separate code file is just making things more confusing IMHO. Could it not be in the NBPhoneNumberUtil file?

Reply:
It certainly could. IMHO, there are a couple of reasons that splitting the files that I think outweighs the confusion this might cause:
1. NBPhoneNumberUtil header is not nullability-annotated, but short number support is properly annotated, which means that to add short number support into existing NBPhoneNumberUtil will require its header to be nullability-annotated which will break the code that's currently using NBPhoneNumberUtil in Swift. (I think in a long run, all header files must be properly annotated, but in the meanwhile, I'd like to reduce breakage as much as possible).

2. Since the canonical version of libPhoneNumber (Java) actually separates short number support into its own ShortNumberInfo class, even though it makes sense for us to put these methods as part of NBPhoneNumberUtil, making it into a separate category will make porting changes a lot easier to deal with.

I think since short number support is optional, and if the user wants it, they already have to add a compiler flag, making the user imports a separate file makes sense to me. Though, I think I should also add instructions in README.md to explain how to use short number support.